src/components/App.tsx

```diff
@@ -1,14 +1,34 @@
 import React from 'react'
 import { hot } from 'react-hot-loader'
-import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
+import { BrowserRouter as Router, Switch, Route, useLocation } from 'react-router-dom'
 import { Grid, Segment } from 'semantic-ui-react'
-
 import ApplicationsList from './ApplicationList'
+import ApplicationNew from './ApplicationNew'
+import Application from './Application'
+import Approval from './Approval'
+import Admin from './Admin'
 import AppMenu from './AppMenu'
 import Footer from './Footer'
-import Register from './Register'
 import Home from './Home'
+import Login from './Login'
+import TemplateList from './TemplateList'
+import TemplateNew from './TemplateNew'
+import Template from './Template'
+import Account from './Account'
+import { Organisation, OrgMemberEdit } from './Organisation'
+import { AdminUsers, AdminPermissions, Config } from './AdminOther'
+import { NotificationsList, Notification } from './Notification'
 import NoMatch from './NoMatch'
+import { ProductList, Product } from './Product'
+
+// queryParams is an object that gets the URL query params as key-value pairs
+// This object should be used for filtering the getApplication query
```

Comment:
I think it's good to have a header for this function, but can this be improved to use a pattern of function header? I'll find one tomorrow if you haven't already. But just to make it easier to find what if you are saying is regarding the result, parameters of the function logic. Also could you change to use ES6 arrow functions instead?

Reply:
Changed to arrow function. Not sure quite what you mean by "function header"? Can you give me an example?

Reply:
The comments before your function. isn't that called a function header? Example of what I would prefer to use:
```
/**
 * Represents a book.
 * @param {string} title - The title of the book.
 * @param {string} author - The author of the book.
 * @returns { (promt:string) => void } - the returned function
 */
```
It seems this is using the JsDocs pattern? I'm not sure what that is, but I thought it would be good to have a pattern for commenting functions. What do you think?

Reply:
Okay, have added something now. Not 100% sure I expressed the return value correctly. Let me know what you think.